mapstory/import_handlers.py

```diff
@@ -51,3 +53,63 @@ def handle(self, layer, layer_config, *args, **kwargs):
                 for date_option in ('start_date', 'end_date'):
                     if layer_config.get(date_option) == date_field:
                         layer_config[date_option] = new_field_name.lower()
+
+
+class GeoserverHandlerMixin(ImportHandlerMixin):
+    """
+    A Mixin for Geoserver handlers.
+    """
+    catalog = gs_catalog
+
+
+class DefaultStyleHandler(GeoserverHandlerMixin):
+    """
+    Apply a default MapStory theme SLD style to imported layer
+    """
+
+    def can_run(self, layer, layer_config, *args, **kwargs):
+        """
+        Only run this handler if the layer is found in Geoserver and the layer's style is one of the default styles.
+        """
+        self.catalog._cache.clear()
+        self.layer = self.catalog.get_layer(layer)
+
+        if self.layer and self.layer.default_style:
+            return self.layer.default_style.name in ['polygon', 'point', 'line']
+        else:
+            return False
+
+    @ensure_can_run
+    def handle(self, layer, layer_config, *args, **kwargs):
+        """
+        Default MapStory style applied depending on feature type
+        """
+
+        # import pydevd
+        # pydevd.settrace('192.168.0.15', port=65432, stdoutToServer=True, stderrToServer=True, trace_only_current_thread=False,overwrite_prev_trace=True,patch_multiprocessing=True)
+
+        sld = self.get_feature_type(self.layer)
+
+        layer = self.catalog.get_layer(self.layer.name)
+        layer.default_style = sld
+        self.catalog.save(layer)
+
+    def get_feature_type(self, layer):
+        """
+        Returns appropriate style to use based on feature type
+        """
+        if layer.resource.resource_type == 'featureType':
+            res = layer.resource
+            res.fetch()
+            ft = res.store.get_resources(res.name)
+            ft.fetch()
+            for attr in ft.dom.find("attributes").getchildren():
+                attr_binding = attr.find("binding")
+                if "jts.geom" in attr_binding.text:
+                    if "Polygon" in attr_binding.text:
+                        name = "Generic_Mapstory_Polygon"
+                    elif "Line" in attr_binding.text:
+                        name = "Generic_Mapstory_Line"
+                    else:
+                        name = "Generic_Mapstory_Point"
+        return name
```

Comment:
might want to return "generic" as the default (if it cannot determine which one it is).
"generic" is a default GS style that should render any layer (but a wee bit inefficiency).

Reply:
added generic style as 'else' case